src/app/services/shopping-cart.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { Book } from './book';
+
+@Injectable()
+export class ShoppingCartService {
+
+  private books = new BehaviorSubject([]);
+  shoppingBooks = this.books.asObservable();
+
+  constructor() {}
+
+  addBokk (book: Book) {
+    this.books.value.push(book);
+    this.books.next(this.books.value);
```

Comment:
```suggestion
    const books = [...this.books.value, {...book}];
    this.books.next(books);
```
this is pure and cute.